packages/design-system-react/src/components/icon/Icon.stories.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useState } from 'react';
+import type { StoryObj } from '@storybook/react';
+import { Icon } from './Icon';
+import { IconName, IconSize, IconColor } from './Icon.types';
+import README from './README.mdx';
+
+const meta = {
+  title: 'React Components/Icon',
+  component: Icon,
+  parameters: {
+    docs: {
+      page: README,
+    },
+  },
+  args: {
+    name: IconName.AddSquare,
+  },
+  argTypes: {
+    name: {
+      description:
+        'Required prop to specify which icon to render from the icon set',
+      control: 'select',
+      options: Object.values(IconName),
+      table: {
+        type: { summary: 'IconName' },
+        defaultValue: { summary: 'required' },
+      },
+    },
+    size: {
+      description: 'Optional prop to control the size of the icon',
+      control: 'select',
+      options: Object.values(IconSize),
+      table: {
+        type: { summary: 'IconSize' },
+        defaultValue: { summary: 'IconSize.Md' },
+      },
+    },
+    color: {
+      description:
+        'Optional prop that sets the color of the icon using predefined theme colors',
+      control: 'select',
+      options: Object.values(IconColor),
+      table: {
+        type: { summary: 'IconColor' },
+        defaultValue: { summary: 'IconColor.IconDefault' },
+      },
+    },
+  },
```

Comment:
Adds controls. Descriptions align with prop type jsDoc comments. Also appears in MDX docs under component api

<img width="1512" alt="Screenshot 2024-11-29 at 10 01 22 AM" src="https://github.com/user-attachments/assets/d43b370f-94a8-4718-bd2c-5f6733e1f94b">
 

<img width="1512" alt="Screenshot 2024-11-29 at 10 00 54 AM" src="https://github.com/user-attachments/assets/40ba5e72-931f-4f82-a995-212d7fa13b4b">

Reply:
- [ ] Add to component template https://github.com/MetaMask/metamask-design-system/pull/139